### DISCUSSION TITLE
LibreChat + agents

### DISCUSSION THREAD

**Discussion by KurtKobalt** [danny-avila/LibreChat#2460] [Q&A]
What is the best approach to extend the functionality of LibreChat? So, let's say I want to integrate it with some agentic framework like crewai or autogen. Should I create a server endpoint completely separate that acts like a man-in-the-middle between librechat and the llm? Or is there any clean an easy way to extend the functionality from librechat itself?

**Comment by weipienlee** [comment]
Same question, although, wouldn't mind to keep it based on LangChain/LangGraph. Are there any plans/roadmaps?

Haven't delved into librechat architecture yet, but what I like from opengpts is that one can potentially hook in any agentic (Lang)Graph and the chat(history), llms, and tools are just "injected". Their approach have other quirks, though. Therefore, I'm eager to see/learn where librechat is heading.

**Reply by danny-avila** [reply]
This discussion is similar to the one here: https://github.com/danny-avila/LibreChat/discussions/2557

I will be adapting something like opengpts but modeled after OpenAI Assistants
